### PR TITLE
Test tweaks following #273

### PR DESCRIPTION
**Pull request by SgtCoDFish**
This adds some checks on the new dummy expired certificate, and changes how we test filtering of expired certs.

I'd rather not check the exact output of `ValidateAndSanitizePEMBundleWithOptions` to check for cert expiry, since that'll be brittle if we change what that function outputs. This test instead explicitly checks for expired certs.

### CODE REVIEW

File: pkg/util/pem_test.go [hunk 1]
```diff
@@ -49,9 +52,10 @@ func TestValidateAndSanitizePEMBundle(t *testing.T) {
 			parts:     []string{dummy.TestCertificate1, string(poisonComment), dummy.TestCertificate2, randomComment, dummy.TestCertificate3, string(poisonComment)},
 			expectErr: false,
 		},
-		"valid bundle with expired cert succeeds": {
-			parts:     []string{dummy.TestCertificate1, dummy.TestExpiredCertificate},
-			expectErr: false,
+		"valid bundle with expired cert succeeds with the expired cert intact": {
+			parts:              []string{dummy.TestCertificate1, dummy.TestExpiredCertificate},
+			expectExpiredCerts: true,
+			expectErr:          false,
```
Comment (erikgb) [inline pkg/util/pem_test.go:58]: Could it make sense to omit fields with the correct value for the test case? I makes things a bit more compact.

**Review by erikgb — APPROVED** [review]
Thanks @SgtCoDFish! This looks good to me. Just a minor question/comment. Feel free to ignore it.

/approve
/lgtm
/hold

File: pkg/util/pem_test.go (hunk 1 above)
Comment (SgtCoDFish) [reply pkg/util/pem_test.go:58]: Could you rephrase that for me? I don't think my morning coffee is kicking in yet and I'm not sure what you mean!

Comment (erikgb) [reply pkg/util/pem_test.go:58]: Sorry! I just meant that the zero value of a `bool` in Golang is `false`. When do we explicitly set at field to the zero value in the test, and when do we not do it? I think the most common is to omit setting the field.

Comment (SgtCoDFish) [reply pkg/util/pem_test.go:58]: Ahhh, I'm with you! It's a fair point.

In tests I like to be explicit about them because that makes the test cases easier to copy+paste and change (although I've not been explicit here).

I should be consistent, but I'm gonna merge this since it won't break anything.

**Comment by SgtCoDFish** [comment]
/unhold

Thanks for the review Erik 😁
